### DISCUSSION TITLE
The python community plugin is archived and no longer maintained by the original maintainer.

### DISCUSSION THREAD

**Discussion by thurahtetaung** [nrwl/nx#12670] [General]
The python community plugin is archived and no longer maintained by the original maintainer. 
If anyone know of an alternative way to get python apps (not django or flask, generic python)
to work in an NX workspace, please let me know.

_Originally posted by @thurahtetaung in https://github.com/nrwl/nx/issues/2482#issuecomment-1280422114_

**Comment by DaSchTour** [comment]
What ware the requirements for the plugin? What does it provide you can't to with run-command?

**Comment by AbdealiLoKo** [comment]
I was looking for whether I can use python with nx too .
And the nx-python git repo - https://github.com/codeonrocks/nx-python seems to have been deleted now...
Some npmjs libraries still seem to exist:
https://www.npmjs.com/package/@nx-python/nx-python
https://www.npmjs.com/package/nx-python

From what I understand, I need to handle atleast the following basic things:
 - install dependencies
 - linting my code
 - (auto)formatting my code
 - testing my code
 - building my code

I could use run-command for linting, formatting, testing. And most likely also building (but i need to think about that a bit more)

But I am not clear on how can I handle dependencies with nx for my multirepo
I understand `pip` does not have 'workspace' support like yarn. (Ref: https://github.com/python-poetry/poetry/issues/2270 )

So, maybe I need to use `poetry` with a plugin like: 
https://pypi.org/project/poetry-workspaces/

Anyone had any success in setting up a python project with nx that I can refer to ?

**Reply by niallmcandrew-work** [reply]
I've recently discovery https://www.npmjs.com/package/@nxlv/python (https://github.com/lucasvieirasilva/nx-plugins) which I have started playing with today and appears to be the best option at the moment. So far I have only used it in test environments to evaluate its usefulness but it looks promising. 

It doesn't have much of a following and only one developer, al be it they have been very responsive to my issues. It would be good to get more awareness of it.

They published a guide yesterday on how to set it up which can be found [here](https://lucasvieirasilva.medium.com/poetry-python-nx-monorepo-5750d8627024). 

FYI I'm not affiliated with the developer or the project myself. I'm just another user looking for a Python plugin to use with NX.
